app/views/Login/index.tsx

```diff
@@ -15,33 +26,108 @@ import {
     requiredStringCondition,
     useForm,
 } from '@togglecorp/toggle-form';
-import {
-    CombinedError,
-    gql,
-} from 'urql';
 
-import UserContext from '#base/context/UserContext';
 import Button from '#components/Button';
+import ButtonLayout from '#components/ButtonLayout';
 import TextInput from '#components/TextInput';
-import { useLoginMutation } from '#generated/types/graphql';
 import useAlert from '#hooks/useAlert';
 import mapSwipeLogo from '#resources/images/mapswipe-logo.svg';
-import {
-    alertCombinedError,
-    checkAndAlertGraphQLResultError,
-} from '#utils/error';
 
 import styles from './styles.module.css';
 
-// eslint-disable-next-line @typescript-eslint/no-unused-vars
-const LOGIN_MUTATION = gql`
-mutation Login($username: String!, $password: String!) {
-    login(username: $username, password: $password) {
-        id
-        displayName
+const APP_ENV = import.meta.env.APP_ENVIRONMENT;
+const COOKIE_NAME = `MAPSWIPE-${APP_ENV}-CSRFTOKEN`;
+const REST_ENDPOINT = import.meta.env.APP_REST_API_DOMAIN;
+const cookies = new Cookies();
+
+async function loginUsingFirebaseToken({ token }: { token: string }) {
+    try {
+        const response = await fetch(`${REST_ENDPOINT}/firebase-auth/`, {
+            method: 'POST',
+            credentials: 'include',
+            headers: {
+                'Content-Type': 'application/json',
+                'X-CSRFToken': cookies.get(COOKIE_NAME),
+            },
+            body: JSON.stringify({ token }),
+        });
+
+        if (!response.ok || response.status !== 200) {
+            let errorMsg = `Request failed with status ${response.status}`;
+            try {
+                const errorData = await response.json();
+
+                if (errorData?.non_field_errors?.length) {
+                    errorMsg = errorData.non_field_errors.join(', ');
+                } else if (typeof errorData === 'object') {
+                    errorMsg = Object.entries(errorData)
+                        .map(([field, messages]) => {
+                            if (Array.isArray(messages)) {
+                                return `${field}: ${messages.join(', ')}`;
+                            }
+                            return `${field}: ${messages}`;
+                        })
+                        .join(' | ');
+                }
+            } catch {
+                // ignore parse errors
+            }
+            throw new Error(errorMsg);
+        }
+
+        return { error: undefined };
+    } catch (err: unknown) {
+        return { error: String(err) || 'Unknown error' };
+    }
+}
+
+// Your Firebase config
+const firebaseConfig = {
+    apiKey: import.meta.env.APP_FIREBASE_API_KEY,
+    authDomain: import.meta.env.APP_FIREBASE_AUTH_DOMAIN,
+    projectId: import.meta.env.APP_FIREBASE_PROJECT_ID,
+};
+
+// Init Firebase
+const app = initializeApp(firebaseConfig);
+const auth = getAuth(app);
```

Comment:
What if these configurations aren't available?

Reply:
What should we do? These should be required for production but not for development.